micro_sam/training/training.py

```diff
@@ -163,28 +170,65 @@ def train_sam(
     else:
         optimizer = torch.optim.Adam(model.parameters(), lr=lr)
 
-    if scheduler is None:
-        scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(
-            optimizer, mode="min", factor=0.9, patience=3, verbose=True
-        )
+    scheduler_params = {"mode": "min", "factor": 0.9, "patience": 3, "verbose": True}
```

Comment:
This will fail if we have a different scheduler class, because it will also get these params.
Let's simplify this:
```
if scheduler_kwargs is None:
  schedulder_kwargs = {...}  # the current defaults
```
Hence, if the user passes the kwargs every option needs to be set.

Reply:
Fixed. Thanks for spotting!